tests/integration/external_provider/runner.py

```diff
@@ -0,0 +1,58 @@
+"""Runner function for SIR model integration tests."""
+
+from typing import Any
+
+import numpy as np
+from numpy.typing import NDArray
+
+from flepimop2.engine import EngineABC
+from flepimop2.system import SystemProtocol
+
+
+def runner(
+    stepper: SystemProtocol,
+    times: NDArray[np.float64],
+    state: NDArray[np.float64],
+    params: dict[str, Any],
+    **kwargs: Any,  # noqa: ARG001
+) -> NDArray[np.float64]:
+    """
+    Simple Euler runner for the SIR model.
+
+    Args:
+        stepper: The system stepper function.
+        times: Array of time points.
+        state: The current state array.
+        params: Additional parameters for the stepper.
+        **kwargs: Additional keyword arguments for the engine. Unused by this runner.
+
+    Returns:
+        The evolved time x state array.
+    """
+    output = np.zeros((len(times), len(state)), dtype=float)
+    output[0] = state
+    for i, t in enumerate(times[1:]):
+        if i == 0:
+            continue
+        dt = t - times[i - 1]
+        dydt = stepper(times[i - 1], output[i - 1], **params)
+        output[i] = output[i - 1] + (dydt * dt)
+    return np.hstack((times.reshape(-1, 1), output))
+
+
+class SirEngine(EngineABC):
+    """SIR model runner."""
+
+    def __init__(self) -> None:
+        """Initialize the SIR runner with the SIR runner function."""
+        super().__init__(runner=runner)
+
+
+def build() -> SirEngine:
```

Comment:
```suggestion
def build(_ : dict[str, Any]) -> EulerEngine:
```

We are likely to check for argument signature at some point on build.

Reply:
I think this required #42 first, since that's been merged the adjustment has been made.